sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManager.kt

```diff
@@ -24,10 +28,30 @@ import com.mapbox.maps.ViewAnnotationOptions
  * by visibility of given feature.
  */
 @MapboxExperimental
-interface ViewAnnotationManager {
+class ViewAnnotationManager(
+  private val mapView: MapView
+) {
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
+  private val viewPlugins = mapView.mapController.pluginRegistry.viewPlugins
+  private val viewAnnotationPositionsUpdateListener = ViewAnnotationPositionsUpdateListener { drawAnnotationViews(it) }
```

Comment:
@mapbox/maps-android that's the only tricky place I believe. We could not make public `ViewAnnotationManager` class implement `ViewAnnotationPositionsUpdateListener` as this will lead to following violation.

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/15800566/153858913-90e2cd2f-f1ec-4b69-b369-6375976473ce.png">